src/common/crypto/platform/openssl/crypto.cpp

```diff
@@ -544,16 +544,54 @@ expected::ExpectedString ExtractPublicKey(const Args &args) {
 	return string(key_vector.begin(), key_vector.end());
 }
 
-expected::ExpectedBytes SignData(const Args &args, const vector<uint8_t> &digest) {
-	auto exp_private_key = PrivateKey::Load(args);
-	if (!exp_private_key) {
-		return expected::unexpected(exp_private_key.error());
+static expected::ExpectedBytes SignED25519(EVP_PKEY *pkey, const vector<uint8_t> &digest) {
```

Comment:
I'm confused, both `SignED25519` and `SignGeneric` take the same input, so why is this called `digest` (`raw_data` is passed in)? `SignGeneric` does perform a digest step internally, but this seems unrelated to what its input argument is.

Maybe a better way is to use the same argument names, but to call the functions `SignMessage` and `SignDigest`, respectively.